src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -150,23 +153,30 @@ protected virtual void SetSubFieldNodes(ExecutionContext context, ObjectExecutio
         Debug.Assert(parent.Result != null || parent is RootExecutionNode);
 
         var parentType = parent.GetObjectGraphType(context.Schema) ?? parent.GraphType;
-        var fields = Interlocked.Exchange(ref context.ReusableFields, null);
-        fields = CollectFieldsFrom(context, parentType, parent.SelectionSet!, fields);
+        var fields = _reusableFields ??= new();
```

Comment:
Will this work better?
```c#
[ThreadStatic]
private static Dictionary<string, (GraphQLField, FieldType)> _reusableFields = new();

protected virtual void SetSubFieldNodes(ExecutionContext context, ObjectExecutionNode parent)
{
    var fields = _reusableFields;
    try
    {
        ...
    }
    finally
    {
        fields.Clear()
    }
}

Reply:
ThreadStatic fields cannot have initializers. It would compile, but they will only run once and all other threads will not be initialized.